_posts/2019-03-22-java-darmowa-czy-nie.md

```diff
@@ -0,0 +1,56 @@
+
+## Oracle JDK vs OpenJDK
+
+W obecnej sytuacji szczególnego znaczenia nabierają różnice pomiędzy JDK releasowanym przez Oracle a OpenJDK.
+* [OpenJDK](http://openjdk.java.net/) jest projektem open source dostarczającym implementację  Java Platform, Standard Edition. Projekt działa od 2007 roku, a jednym z głównych kontrybutorów jest Oracle.
+* [Oracle JDK](https://www.oracle.com/technetwork/java/javase/overview/index.html) natomiast to implementacja Java Platform dostarczana i supportowana przez Oracle w ramach OTN (Oracle Technology Network).
+
+Obydwie wersję mają w zasadzie ten sam code base. Techniczne różnice są niewielkie i dotyczą głównie narzędzi i deploymentu. Różnice, które z punktu widzenia ostatnich zmian są najbardziej istotnie to licencjonowanie i cykl wydawniczy.
+
+## Licencjonowanie i opłaty
+
+OpenJDK jest projektem open source licencjonowanym w oparciu o GNU General Public License, version 2 with CE.
+
+W Oracle JDK model licencjonowania różni się w zależności od wersji. Wersje z przed wersji 11 licencjonowane są w oparciu o [Oracle BCL (Binary Code Licence)](https://www.oracle.com/technetwork/java/javase/terms/license/index.html). Od wersji 11 Oracle JDK licencjonowane jest w oparciu o [Oracle Java SE OTN License](https://www.oracle.com/technetwork/java/javase/terms/license/javase-license.html), **który nie pozwala na komercyjne użycie**. Jeżeli chcemy używać tej wersji do zastosowań komercyjnych musimy wykupić subskrypcję "[Oracle Java SE subscription](https://www.oracle.com/java/java-se-subscription.html)". Subskrypcja jest rozliczana w cyklu miesięcznym, a końcowa cena zależy od:
+* sposobu użycia (desktop/serwer)
+* liczby rdzeni
+* mnożnika [Oracle Processor Core Factor](http://www.oracle.com/us/corporate/contracts/processor-core-factor-table-070634.pdf)
+
+Wg. aktualnego [cennika](https://www.oracle.com/assets/java-se-subscription-pricelist-5028356.pdf) za jeden obliczony na podstawie mnożnika core zapłacimy 25$ przy założeniu, że corów jest mnie niż 99.
+
+## Cykl wydawniczy
+
+Biorąc pod uwagę powyższe informacje wydaje się, że naturalnym krokiem dla tych, którzy nie chcą płacić za support jest przesiadka na wersję OpenJDK, jest tu jednak pewien haczyk. Aby go odkryć musimy wiedzieć jaki jest cykl wydawniczy poszczególnych wersji.
+
+Jakiś czas temu Oracle postanowił nadać pewien rygor czasowy kolejnym wydawanym wersjom. Kolejne wersje Javy będą ukazywały się co pół roku. Niektóre z nich będą wersjami LTS, a pozostałe będą zastępowane kolejnymi i nie będą dalej rozwijane. Aktualnie wersje LTS to:
+* wersja 8 - wspierana do marca 2025 roku,
+* wersja 11 - wspierana do września 2026 roku.
+
+Oracle będzie publikował nowe wydania wersji raz na kwartał. Dotyczy to zarówno wersji LTS jak i nie LTS. Niestety w przypadku OpenJDK Oracle nie będzie wydawał więcej niż dwóch wersji również dla wersji LTS. Oznacza to, że jeżeli wybieramy wersję LTS oczekując długiego czasu wsparcia nie możemy polegać na OpenJDK. 
```

Comment:
"Oznacza to, że jeżeli wybieramy wersję LTS oczekując długiego czasu wsparcia nie możemy polegać na OpenJDK." IMO ten akapit nie powinien się kończyć tym zdaniem, zwłaszcza, że niżej podajemy linki do OpenJDK i wręcz piszemy, że taki RedHat będzie jeszcze kilka lat wspierał LTSy.